rts/Game/UI/GuiHandler.cpp

```diff
@@ -3936,7 +3936,8 @@ void CGuiHandler::DrawMapStuff(bool onMiniMap)
 		assert(buildeeDef != nullptr);
 
 		if (!buildInfoQueue.empty()) {
-			unitDrawer->SetupShowUnitBuildSquares(onMiniMap, true);
+			if (drawBuildGrid || onMiniMap)
+				unitDrawer->SetupShowUnitBuildSquares(onMiniMap, true);
```

Comment:
this isn't your fault, but SetupShowUnitBuildSquares is really an awful function: the second argument is always known at compile time so we can easily avoid unnecessary calls entirely, and the first parameter is not actually used in the function..

Reply:
I lack the knowledge yet to better that function; however about adding 'onMinimap' it's the same as in maintenance version. It is needed to distinguish minimap grid drawing and main window drawing.